x-pack/plugins/security_solution/public/common/components/query_bar/index.tsx

```diff
@@ -102,7 +106,26 @@ export const QueryBar = memo<QueryBarComponentProps>(
       [filterManager]
     );
 
-    const indexPatterns = useMemo(() => [indexPattern], [indexPattern]);
+    const dataView = useMemo(() => {
+      if (Object.hasOwn(indexPattern, 'getName')) {
+        return [indexPattern] as DataView[];
+      }
+      const dv = new DataView({
+        spec: {
+          ...indexPattern,
+          fields: indexPattern.fields.reduce((acc, field) => {
+            // @ts-expect-error missing 'searchable' , aggregatable properties
+            // this is okay because it's better than having the whole app
+            // explode
+            acc[field.name] = field;
+            return acc;
+          }, {} as DataViewFieldMap),
+        },
+        fieldFormats,
+      });
+      return [dv];
+    }, [indexPattern, fieldFormats]);
```

Comment:
Thanks, Devin, for sharing the second option, Personally I think the second option makes more sense since I imagine it could be reusable if the user wanted to use it anywhere else, does it make sense?